detectable/src/main/java/com/blackduck/integration/detectable/detectables/opam/buildexe/parse/OpamTreeParser.java

```diff
@@ -9,141 +9,99 @@
 import com.blackduck.integration.bdio.model.externalid.ExternalId;
 import com.blackduck.integration.bdio.model.externalid.ExternalIdFactory;
 import com.blackduck.integration.detectable.detectable.codelocation.CodeLocation;
-import com.blackduck.integration.detectable.detectable.util.DetectableStringUtils;
-import org.apache.commons.lang3.StringUtils;
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.util.List;
 import java.util.ArrayList;
-import java.util.Stack;
 import java.io.File;
 
 public class OpamTreeParser {
 
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
-    private int level = 0;
-    private String currentProject;
-    private static final String DIRECT_DEPENDENCY_INDICATOR = "|--"; // direct dependencies usually start with this prefix
-    private static final String[] DEPENDENCY_INDICATOR = new String[] { "|", "|--", "'--" }; // indicates that line contains a dependency
-    private static final String PACKAGE_PREFIX = "-- ";
-    private final List<OpamParsedResult> codeLocations = new ArrayList<>();
+    private List<OpamParsedResult> codeLocations = new ArrayList<>();
     private DependencyGraph currentGraph;
-    private Stack<Dependency> dependencyStack;
     private final File sourceDirectory;
     private final ExternalIdFactory externalIdFactory;
+    private final Gson gson;
 
-    public OpamTreeParser(File sourceDirectory, ExternalIdFactory externalIdFactory) {
+    public OpamTreeParser(Gson gson, File sourceDirectory, ExternalIdFactory externalIdFactory) {
+        this.gson = gson;
         this.sourceDirectory = sourceDirectory;
         this.externalIdFactory = externalIdFactory;
     }
 
-    public List<OpamParsedResult> parseTree(List<String> tree) {
-        currentProject = null;
-        currentGraph = new BasicDependencyGraph();
-        dependencyStack = new Stack<>();
-
+    public List<OpamParsedResult> parseJsonTreeFile(File outputFile) throws IOException {
+        if(outputFile == null || !outputFile.exists() || !outputFile.isFile()) {
+            logger.warn("The opamTreeOutput.json file does not exist, the file was deleted, or an issue occurred while running the opam tree command.");
+            return codeLocations;
+        }
 
-        for(String line: tree) {
-            if(line.isEmpty() || line.startsWith("*") || line.startsWith("=")) {
-                continue;
+        try (JsonReader jsonReader = new JsonReader(new FileReader(outputFile))) {
+            OpamJsonFileModule opamTreeModule = gson.fromJson(jsonReader, OpamJsonFileModule.class);
+            for(OpamTreeProjectModule opamTreeProjectModule: opamTreeModule.opamProjects) {
+                if(opamTreeProjectModule != null) {
```

Comment:
I think you can remove this check, or the one in processTreeJsonObject, as the first thing that method does is check for the null so we are checking it twice. I'd say for code readability you could just call the method and let it be responsible for determining if it can operate on the object.